src/gausskernel/storage/access/k2/catalog/sql_catalog_client.cpp

```diff
@@ -20,87 +20,78 @@ Copyright(c) 2022 Futurewei Cloud
-  CreateDatabaseRequest request {
-    .databaseName = database_name,
-    .databaseId = database_id,
-    .databaseOid = database_oid,
-    .sourceDatabaseId = source_database_id,
-    .creatorRoleName = creator_role_name,
-    .nextPgOid = next_pg_oid
-  };
-  auto start = k2::Clock::now();
-  auto result = catalog_manager_->CreateDatabase(request);
-  K2LOG_D(log::catalog, "CreateDatabase {} took {}", database_name, k2::Clock::now() - start);
-  return std::get<0>(result);
+    CreateDatabaseRequest request {
+      .databaseName = database_name,
+      .databaseId = database_id,
+      .databaseOid = database_oid,
+      .sourceDatabaseId = source_database_id,
+      .creatorRoleName = creator_role_name,
+      .nextPgOid = next_pg_oid
+    };
+    auto result = catalog_manager_->CreateDatabase(request);
+    return k2pg::K2StatusToK2PgStatus(std::move(std::get<0>(result)));
 }
 
-sh::Status SqlCatalogClient::DeleteDatabase(const std::string& database_name, const std::string& database_id) {
-  return catalog_manager_->DeleteDatabase(database_name, database_id);
+Status SqlCatalogClient::DeleteDatabase(const std::string& database_name, const std::string& database_id) {
+    return k2pg::K2StatusToK2PgStatus(catalog_manager_->DeleteDatabase(database_name, database_id));
 }
 
-sh::Status SqlCatalogClient::UseDatabase(const std::string& database_name) {
-  auto start = k2::Clock::now();
-  auto status = catalog_manager_->UseDatabase(database_name);
-  K2LOG_D(log::catalog, "UseDatabase {} took {}", database_name, k2::Clock::now() - start);
-  return status;
+Status SqlCatalogClient::UseDatabase(const std::string& database_name) {
+    return k2pg::K2StatusToK2PgStatus(catalog_manager_->UseDatabase(database_name));
 }
 
-sh::Status SqlCatalogClient::CreateTable(
+Status SqlCatalogClient::CreateTable(
     const std::string& database_name,
     const std::string& table_name,
     const PgObjectId& table_object_id,
     PgSchema& schema,
     bool is_pg_catalog_table,
     bool is_shared_table,
     bool if_not_exist) {
-  CreateTableRequest request {
-    .databaseName = database_name,
-    .databaseOid = table_object_id.GetDatabaseOid(),
-    .tableName = table_name,
-    .tableOid = table_object_id.GetObjectOid(),
-    .schema = schema,
-    .isSysCatalogTable = is_pg_catalog_table,
-    .isSharedTable = is_shared_table,
-    .isNotExist = if_not_exist
-  };
-  auto start = k2::Clock::now();
-  auto response = catalog_manager_->CreateTable(request);
-  K2LOG_D(log::catalog, "CreateTable {} in {} took {}", table_name, database_name, k2::Clock::now() - start);
-  return std::get<0>(response);
+    CreateTableRequest request {
+      .databaseName = database_name,
+      .databaseOid = table_object_id.GetDatabaseOid(),
+      .tableName = table_name,
+      .tableOid = table_object_id.GetObjectOid(),
+      .schema = schema,
+      .isSysCatalogTable = is_pg_catalog_table,
+      .isSharedTable = is_shared_table,
```

Comment:
can we remove is shared table?

Reply:
No, we cannot remove is_shared_table since a shared table might be stored on a different k2 collection from the current database collection. Catalog manager uses this flag to redirect a shared table to the correct k2 collection.